src/components/Home/Shop.js

```diff
@@ -1,17 +1,25 @@
 import { StarIcon } from '@heroicons/react/solid'
-import React from 'react'
+import React, { useCallback } from 'react'
 import { Link } from 'react-router-dom'
 
-function Shop ({ name, address }) {
+function Shop ({ address, loading, name }) {
+  const linkDisableHandler = useCallback(
```

Comment:
Shouldn't this be called `linkClickHandler` or `shopClickHandler`?

Reply:
This is for disabling the link.

Reply:
Yes but only when in loading state otherwise it is working as usual?

Reply:
Yes

Reply:
It is for disabling the Link when shop component is in loading state